### DISCUSSION TITLE
Change default mega-linter.yml in own dockerfile

### DISCUSSION THREAD

**Discussion by yannik1999** [oxsecurity/megalinter#2643] [Q&A] [answered]
I'm currently building an own mega-linter dockerimage. I want to change the .mega-linter.yml default config in the dockerimage. Where do I have to copy my .mega-linter.yml to achieve this? 

I already exchanged the other standard linter configs in the /action/lib/.automation/ dir. This does not work for the .mega-linter.yml.

Can someone help me here please. Thank you :)

**Comment by nvuillam** [comment]
There is no default .mega-linter.yml, as it is defined in the repositories MegaLinter is running on :)

Why not just changing the default linter configs just for your repositories ?

Or maybe you'd like an option to be able to dynamically import external default configurations ?

I'd prefer we find a solution using official MegaLinter image, what is the use case behind what you are trying to achieve ? :)

**Reply by yannik1999** [reply]
Hello nvuillam,

first, thank you for the fast answer. 

What I try to achieve is to make this linter as easy usable as possible in different project in our company:

I created also different flavors, as you did, and gave them default linter configs, which fit to our needs. Those are our dockerimages based on the megalinter flavors. I also created a job template for Azure DevOPs Pipelines which is nearly the same as your pipeline for azure pipelines mention in the documentation. 
If I am now able to also give in another default .mega-linter.yml inside the dockerimage, others just need to add the template of the right flavor inside their pipeline, and they are finished. No extra configs inside the project and very easy to use with nearly zero effort and knowledge about the linter. Even if they want to change something, they still can do this.

**Reply by nvuillam** [reply]
If you need new default official flavors we can eventually discuss about it, maybe new flavors could be compliant to certain types of repos :)

If you want to keep generating your own images, you can create a public repo and add its path in your dockerfile : 

`ENV MEGALINTER_CONFIG=https://raw.githubusercontent.com/tcjoel/megalinter-config/main/mega/.mega-linter.yml`

**Reply by nvuillam** [reply, marked as answer]
After this PR is merged , you'll be able to include MegaLinter config file in your image then use

`ENV MEGALINTER_CONFIG=/path/to/your/.mega-linter.yml`

https://github.com/oxsecurity/megalinter/pull/2649

If it's not confidential, where do you work ?

You can add me on LinkedIn to follow the latest news about MegaLinter :)

https://www.linkedin.com/in/nicolas-vuillamy/

**Reply by yannik1999** [reply]
Thank you very much for your effort. Already tried it and it works. 

You already added me on LinkedIn and I accepted you. I appreciate your work. 

When do you plan the next release? I used the Beta Tags of the dockerimages, but having it as real release would be even better. I heard you are planning to release v7?

**Reply by nvuillam** [reply]
About v7 release... it's a matter of weeks:)
